internal/wasm/store_test.go

```diff
@@ -86,48 +85,84 @@ func TestModuleInstance_Memory(t *testing.T) {
 	}
 }
 
-func TestStore_AddHostFunction(t *testing.T) {
+func TestStore_NewHostModule(t *testing.T) {
 	s := newStore()
 
-	hf, err := NewGoFunc("fn", func(wasm.ModuleContext) {})
-	require.NoError(t, err)
-
 	// Add the host module
-	hostModule := &ModuleInstance{Name: "test", Exports: make(map[string]*ExportInstance, 1)}
-	s.moduleInstances[hostModule.Name] = hostModule
-
-	_, err = s.AddHostFunction(hostModule, hf)
+	_, err := s.NewHostModule("test", map[string]interface{}{"fn": func(wasm.ModuleContext) {}})
 	require.NoError(t, err)
 
+	// Ensure it was added to module instances
+	hm := s.moduleInstances["test"]
+	require.NotNil(t, hm)
+
 	// The function was added to the store, prefixed by the owning module name
 	require.Equal(t, 1, len(s.functions))
 	fn := s.functions[0]
 	require.Equal(t, "test.fn", fn.Name)
 
 	// The function was exported in the module
-	require.Equal(t, 1, len(hostModule.Exports))
-	exp, ok := hostModule.Exports["fn"]
+	require.Equal(t, 1, len(hm.Exports))
+	_, ok := hm.Exports["fn"]
 	require.True(t, ok)
 
 	// Trying to register it again should fail
-	_, err = s.AddHostFunction(hostModule, hf)
-	require.EqualError(t, err, `"fn" is already exported in module "test"`)
+	_, err = s.NewHostModule("test", map[string]interface{}{"host_fn": func(wasm.ModuleContext) {}})
+	require.EqualError(t, err, "module test has already been instantiated")
+}
 
-	// Any side effects should be reverted
-	require.Equal(t, []*FunctionInstance{fn, nil}, s.functions)
-	require.Equal(t, map[string]*ExportInstance{"fn": exp}, hostModule.Exports)
+func TestPublicModule_String(t *testing.T) {
+	s := newStore()
+
+	// Ensure paths that can create the host module can see the name.
+	m, err := s.Instantiate(&Module{}, "module")
+	require.NoError(t, err)
+	require.Equal(t, "Module[module]", m.String())
+	require.Equal(t, "Module[module]", s.Module(m.instance.Name).String())
 }
 
-func TestStore_ExportImportedHostFunction(t *testing.T) {
+func TestHostModule_String(t *testing.T) {
```

Comment:
same goes for this

Reply:
same here, this is making sure store which is instantiating it, is setting the name field..

Reply:
same .. I forgot even though these are functions with store pointers that do the stuff, it is in host.go